tests/backend/test_bosonic_backend.py

```diff
@@ -611,16 +611,16 @@ def test_measurement(self, alpha, phi, r):
             sf.ops.MeasureX | q[0]
             sf.ops.MeasureHD | q[1]
         backend = bosonic.BosonicBackend()
-        applied, samples, all_samples = backend.run_prog(prog)
+        applied, samples_dict = backend.run_prog(prog)
         state = backend.state()
 
         # Check output is vacuum since everything was measured
         assert np.allclose(state.fidelity_vacuum(), 1)
 
         # Check samples
         for i in range(2):
-            assert i in samples.keys()
-            assert samples[i].shape == (1,)
+            assert i in samples_dict.keys()
+            assert samples_dict[i].shape == (1, 1)
```

Comment:
Why did this shape change?

Reply:
It had to do with a NumPy array casting taking place, which I've now changed. Although I was also a bit confused when looking at this. There was some strange array dimension changes going on in places.